RustLeanModels/Basic.lean

```diff
@@ -52,44 +55,61 @@ theorem Eq_iff_of_unique_and_mp : unique p → (∀ y, (x = y) → p y) → (∀
   unfold unique at g1; exact g1 x y g2 g3
 
 @[simp]
-lemma not_sym :  ¬ x = y →  ¬ y = x := by intro g; by_contra; rename_i g1; symm at g1; contradiction
+theorem not_sym :  ¬ x = y →  ¬ y = x := by intro g; intro h; have h := Eq.symm h; contradiction
 
-lemma exists_EQ {α: Type} {p q : α → Prop}: (∀ x, p x  ↔ q x) → ((∃ y, p y) ↔ (∃ y, q y)) :=by
+theorem exists_EQ {α: Type} {p q : α → Prop}: (∀ x, p x  ↔ q x) → ((∃ y, p y) ↔ (∃ y, q y)) :=by
   intro g; constructor
-  intro gy ; obtain ⟨y,gy⟩ := gy ; rw[g y] at gy ; use y
-  intro gy ; obtain ⟨y,gy⟩ := gy ; rw[← g y] at gy ; use y
-
-lemma not_in_forall_not_eq {s: List α} (h: ∀ x ∈ s, a ≠ x) : a ∉ s :=by
-  induction s; simp; rename_i head tail ind
```

Comment:
induction is overkill here. We could do this instead:
```
theorem not_in_forall_not_eq {s: List α} (h: ∀ x ∈ s, a ≠ x) : a ∉ s := by  
  intro g; have := h a g; contradiction
```

Reply:
Nice!